src/Microsoft.OData.Client/Materialization/MaterializerDeltaResourceSet.cs

```diff
@@ -0,0 +1,97 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MaterializerDeltaResourceSet.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.OData.Client.Materialization
+{
+    /// <summary>
+    /// Materializer state for a given <see cref="ODataDeltaResourceSet"/>.
+    /// </summary>
+    internal class MaterializerDeltaResourceSet : IMaterializerState
+    {
+        /// <summary>The OData delta resource set.</summary>
+        private readonly ODataDeltaResourceSet deltaResourceSet;
+
+        /// <summary>The entries.</summary>
+        private readonly List<IMaterializerState> entries;
+
+        /// <summary>
+        /// Prevents a default instance of the <see cref="MaterializerDeltaResourceSet"/> struct from being created.

```

Comment:
```suggestion
        /// Prevents a default instance of the <see cref="MaterializerDeltaResourceSet"/> class from being created.

```
Is that the primary role of this constructor?

Reply:
yes. and also creating an instance of `MaterializerDeltaResourceSet` within the class.